### ISSUE TITLE
For mmrotate tensorrt int8 model, I met this problem.

### ISSUE THREAD

**Issue by JinqingZhengTju** [open-mmlab/mmdeploy#1048]
When I set int8 model for mmrotate, I met this problem as follows. I don't know the reason.
0%|          | 0/40 [00:00<?, ?it/s][2022-09-14 22:52:07.137] [mmdeploy] [info] [model.cpp:95] Register 'DirectoryModel'
  0%|          | 0/40 [00:01<?, ?it/s]
Process Process-3:
Traceback (most recent call last):
  File "/home/zjq/anaconda3/envs/open-mmlab/lib/python3.7/multiprocessing/process.py", line 297, in _bootstrap
    self.run()
  File "/home/zjq/anaconda3/envs/open-mmlab/lib/python3.7/multiprocessing/process.py", line 99, in run
    self._target(*self._args, **self._kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/core/pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/calibration.py", line 82, in create_calib_input_data
    device=device)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/core/pipeline_manager.py", line 356, in _wrap
    return self.call_function(func_name_, *args, **kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/core/pipeline_manager.py", line 326, in call_function
    return self.call_function_local(func_name, *args, **kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/core/pipeline_manager.py", line 275, in call_function_local
    return pipe_caller(*args, **kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/core/pipeline_manager.py", line 107, in __call__
    ret = func(*args, **kwargs)
  File "/home/zjq/open-mmlab/MMDeploy/mmdeploy/apis/utils/calibration.py", line 67, in create_calib_input_data
    input_ndarray = input_tensor.detach().cpu().numpy()
AttributeError: 'DataContainer' object has no attribute 'detach'
2022-09-14 22:52:08,470 - mmdeploy - ERROR - `mmdeploy.apis.calibration.create_calib_input_data` with Call id: 1 failed. exit.

**Comment by tpoisonooo** [comment]
@lvhan028

**Comment by JinqingZhengTju** [comment]
> @lvhan028

The calibration data is the validation dataset. But the error happened when calling the function 'create_calib_input_data'.

**Comment by grimoire** [comment]
`'DataContainer' object has no attribute 'detach'` can be fixed by update `get_tensor_from_input` in `mmdeploy/codebase/mmrotate/deploy/rotated_detection.py`

```python

    @staticmethod
    def get_tensor_from_input(input_data: Dict[str, Any]) -> torch.Tensor:
        """Get input tensor from input data.

        Args:
            input_data (dict): Input data containing meta info and image
                tensor.
        Returns:
            torch.Tensor: An image in `Tensor`.
        """
        img_data = input_data['img'][0]
        if isinstance(img_data, DataContainer):
            return img_data.data[0]
        return img_data
```

But it seems there are still problems. Still working on it.

**Comment by grimoire** [comment]
Hi, I have a fix in https://github.com/open-mmlab/mmdeploy/pull/1050. Please have a try.
Please note that **DO NOT** enable fp16 mode in the deploy config.

**Comment by JinqingZhengTju** [comment]
> Hi, I have a fix in #1050. Please have a try. Please note that **DO NOT** enable fp16 mode in the deploy config.

Thanks for the solution. I have solved this problem with this method which you provided.
